### ISSUE TITLE
Cookie Banner Accessibility update from 5/12

### ISSUE THREAD

**Issue by edthedev** [app-illinois/Design-Resources#155]
## Context

Further accessibility verification.

## Details

The banner needs to implement the ARIA [Alert Dialog Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/). Currently, the banner is missing role=”alertdialog” on the banner container (id=”ilaCookieBFocusOnLoad”) and it needs to trap focus per the ARIA [Modal Dialog Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/).
        Focus must be managed when the cookie notice is dismissed. The reason for this is that screen reader behavior is inconsistent when the item with focus is hidden or removed from the DOM—in some instances it will crash the screen reader. When the dialog is dismissed, focus should be placed on the body the page in this instance.
    A <br> tag is used to separate the Cookie Notice explanatory text and the link to the cookie policy. The break tag does not cause screen readers to pause when they are reading a line of text. As implemented, there will be a slight pause (length depending on the screen reader in use) at the end of the sentence followed by the announcement of the link. If that is desired in order to ensure that people know that the link is there, then it’s fine to leave it as is. That being said, it may be better to close out the paragraph of text and then include the link as a new paragraph of text.

## Tasks

- [x] update dialog aria role
- [x] place the focus back on the page body when the banner is dismissed
- [x] remove break before cookie detail link
- [x]  The alert role should be removed from the banner parent div (class=”cookie-holder”). That will be handled by the alertdialog role on the banner itself.
- [x] The aria-label in the banner container should be changed to an aria-labelledby, and its value should point to the ID of the visible heading in the banner content (You’ll need to add an id to that heading to point to). The reason for this is that aria-label should only be used when there is no visible label in the page to point to.
- [x]   Focus should be placed on the first focusable item within the banner rather than on the banner itself. That is according to the more recent pattern for modal dialogs, and I verified that the user experience is less twitchy when that is done. This also means that you can remove the tabindex from the banner container.

**Comment by edthedev** [comment]
Further testing revealed: 

    The alert role should be removed from the banner parent div (class=”cookie-holder”). That will be handled by the alertdialog role on the banner itself.
    The banner container (id=”ilaCookieBFocusOnLoad”) also needs aria-model=”true” placed on it.
    The aria-label in the banner container should be changed to an aria-labelledby, and its value should point to the ID of the visible heading in the banner content (You’ll need to add an id to that heading to point to). The reason for this is that aria-label should only be used when there is no visible label in the page to point to.
    Focus should be placed on the first focusable item within the banner rather than on the banner itself. That is according to the more recent pattern for modal dialogs, and I verified that the user experience is less twitchy when that is done.
        This also means that you can remove the tabindex from the banner container.

**Comment by edthedev** [comment]
Our placing the focus back depends on an H1 tag being present, which it should anyway.
In our testing, skipTo.js tends to get the next focus, when navigating by keyboard, which seems acceptable.

**Comment by edthedev** [comment]
Should discuss modal against at Friday's meeting, since we will have both accessibility specialists and Privacy specialists available.
